### ISSUE TITLE
Same image repo & tag for two different skaffold configs.

### ISSUE THREAD

**Issue by iosifnicolae2** [GoogleContainerTools/skaffold#8008]
### Expected behavior
I expected each generated helm manifest to have the right image repo & tag.

### Actual behavior
When using `requires`, skaffold uses the first image when generating the helm manifests.

### Information

- Skaffold version: v2.0.1
- Operating system: MacOS Ventura
- Installed via: Homebrew

### Steps to reproduce the behavior

1. `git clone https://github.com/iosifnicolae2/skaffold-bug && cd skaffold-bug`
1. `make render` - [generated output](https://gist.githubusercontent.com/iosifnicolae2/9f9606f79919858a61c4e817beb97b0e/raw/8c719e3ea07bd1faa604d46a02d842293ebefc21/gistfile1.txt)
1.  You will notice that the two deployments have the same image name & tag
<img width="1156" alt="image" src="https://user-images.githubusercontent.com/7108834/198997622-8c46c31c-6441-4777-8ec4-1d07207d2736.png">

Note. If we execute skaffold in each directory( `skaffold/app1` and `skaffold/app2`) everything works correctly.

I've attached skaffold logs [here](https://gist.githubusercontent.com/iosifnicolae2/2dfac23103a166252d33c00869969f91/raw/2279261fe6ed259266afb1d9f70ff65a34bbc006/gistfile1.txt).

**Comment by iosifnicolae2** [comment]
Duplicate of https://github.com/GoogleContainerTools/skaffold/issues/5317

**Comment by aaron-prindle** [comment]
Closing as duplicate of #5317

**Comment by aaron-prindle** [comment]
Thanks for the reproduction logic here, I was not fully aware of #5317 when I was working on the inital fix and our tests do not have multi-config/multiple-artifact-overrides samples so this was missed in testing.  The solution for now is to not use `setValueTemplates` but `setValues` which doesn't have this issue.  `setValueTemplates` was used initially as there are more template values for things like `IMAGE_DOMAIN` which  imageStrategy.helm+explicitRegistry required.  Now this is supported via another method + `setValues` in #8064

**Comment by aaron-prindle** [comment]
@iosifnicolae2 I have submitted #8066 to make our automatic upgrade logic work correctly here.  You can actually fix this on your end in the short term though.  Essentially using `setValueTemplates` is not desirable/feasible.  What should be done is using `setValues` replacing all image related tag values with the artifact name.  For the yaml files in the `skaffold-bug` repo, this looks like:

```
apiVersion: skaffold/v3
kind: Config
build:
  artifacts:
    - context: ../../app1/
      image: app1
  tagPolicy:
    inputDigest: {}
deploy:
  helm: {}
manifests:
  helm:
    releases:
      - chartPath: ../../charts/app
        createNamespace: true
        name: app1
        setValues:
          image.repository: app1
          image.tag: app1
```
```
apiVersion: skaffold/v3
kind: Config
build:
  artifacts:
    - context: ../../app2/
      image: app2
  tagPolicy:
    inputDigest: {}
deploy:
  helm: {}
manifests:
  helm:
    releases:
      - chartPath: ../../charts/app
        createNamespace: true
        name: app1
        setValues:
          image.repository: app2
          image.tag: app2
```

**Comment by aaron-prindle** [comment]
Fixed with #8066, closing
